### PR TITLE
[WIP] PIM-5853: ajaxify the export builder content tab (rebased PR)

### PR DESCRIPTION
**Pull request by phaseinducer**
Rebase & rework PR for POC https://github.com/akeneo/pim-community-dev/pull/4613

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/export/product/edit/content.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                mediator.clear('pim_enrich:form');
+                Backbone.Router.prototype.once('route', this.unbindEvents);
+
+                if (_.has(module.config(), 'forwarded-events')) {
+                    this.forwardMediatorEvents(module.config()['forwarded-events']);
+                }
+
+                return BaseForm.prototype.configure.apply(this, arguments);
+            },
+            render: function () {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/export/product/edit/content.js]: jsdoc

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/export/product/edit/content.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        _,
+        Backbone,
+        template,
+        BaseForm,
+        mediator,
+        FetcherRegistry
+    ) {
+        return BaseForm.extend({
+            template: _.template(template),
+            configure: function () {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/export/product/edit/content.js]: jsdoc

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/export/product/edit/content.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                }
+                this.getRoot().trigger('pim_enrich:form:render:before');
+
+                this.$el.html(
+                    this.template({})
+                );
+
+                this.renderExtensions();
+            },
+            unbindEvents: function () {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/export/product/edit/content.js]: jsdoc

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/export/product/edit/content.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                mediator.clear('pim_enrich:form');
+            },
+            clearCache: function () {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/export/product/edit/content.js]: jsdoc

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/filter.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    return BaseForm.extend({
+        className: 'control-group',
+        elements: {},
+        removable: false,
+        filterTemplate: _.template(filterTemplate),
+        events: {
+            'click .remove': 'removeFilter'
+        },
+
+        setField: function (field) {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/filter.js]: jsdoc

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/product/completeness.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+], function (_, __, BaseFilter, Routing, template) {
+    return BaseFilter.extend({
+        template: _.template(template),
+        events: {
+            'change [name="filter-value"]': 'updateState'
+        },
+        initialize: function (config) {
+            this.config = config.config;
+        },
+        renderInput: function () {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/product/completeness.js]: jsdoc

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/product/completeness.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        updateState: function () {
+            var value = this.$('[name="filter-value"]').val();
+
+            if ('all' === value) {
+                this.clearData();
+
+                return;
+            }
+
+            this.setData({
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/filter/product/completeness.js]: change after completeness merge

File: features/export/edit_an_export.feature [hunk 8]
```diff
@@ -28,12 +28,13 @@ Feature: Edit an export
...
-    Then I should see the Channel, Status, Completeness, Family, Locales fields
+    Then I should see the Channel, Locales fields
+    Then I should see the filters enabled, completeness, updated and family.code
     And I fill in the following information:
       | Channel      | Tablet                               |
-      | Status       | Disabled                             |
-      | Family       | Boots                                |
-      | Completeness | Not complete on all selected locales |
+    Then I filter by "enabled" with operator "=" and value "false"
+    Then I filter by "family.code" with operator "In list" and value "Boots"
```
Comment (juliensnz) [inline features/export/edit_an_export.feature]: and

File: features/export/edit_an_export.feature [hunk 9]
```diff
@@ -28,12 +28,13 @@ Feature: Edit an export
       | Date format       | yyyy-MM-dd |
     And I uncheck the "With header" switch
     When I visit the "Content" tab
-    Then I should see the Channel, Status, Completeness, Family, Locales fields
+    Then I should see the Channel, Locales fields
+    Then I should see the filters enabled, completeness, updated and family.code
```
Comment (juliensnz) [inline features/export/edit_an_export.feature]: and

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/export/product/edit/content/data.html [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<div class="accordion-heading">
+    <a class="accordion-toggle" data-toggle="collapse" href="#collapse-3-2">
+        <i class="icon-chevron-down"></i>
+        Data filters
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/export/product/edit/content/data.html]: use translations

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/export/product/edit/content/data.html [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+        <div data-drop-zone="headings" style="float: right; position: relative; top: -12px;"></div>
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/export/product/edit/content/data.html]: change style

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/export/product/edit/content/structure.html [hunk 12]
```diff
@@ -0,0 +1,13 @@
+<div class="accordion-heading">
+    <a class="accordion-toggle" data-toggle="collapse" href="#collapse-3-1">
+        <i class="icon-chevron-down"></i>
+        Structure filters
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/export/product/edit/content/structure.html]: Add translations

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/export/product/edit/content/structure.html [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+    </a>
+</div>
+<div id="collapse-3-1" class="accordion-body in">
+    <div class="accordion-inner">
+        <div class="control-group" data-drop-zone="filters" style="width: 600px;"></div>
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/export/product/edit/content/structure.html]: remove style

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/export/product/edit/content/structure/attributes.html [hunk 14]
```diff
@@ -0,0 +1,4 @@
+<label class="control-label required">Attributes</label>
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/export/product/edit/content/structure/attributes.html]: translations

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/filter/product/category.html [hunk 15]
```diff
@@ -1,8 +1,8 @@
-<div class="flex-space-between input-like<% if (disabled) { %> disabled<% } %>">
+<div class="flex-space-between input-like <% !isEditable ? 'disabled' : '' %>">
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/filter/product/category.html]: <%-

File: src/Pim/Bundle/EnrichBundle/Resources/public/templates/filter/simpleselect.html [hunk 16]
```diff
@@ -0,0 +1,11 @@
+<label class="control-label required"><%- field %></label>
+<div class="controls" style="width: 600px;">
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/templates/filter/simpleselect.html]: remove style

File: src/Pim/Bundle/ImportExportBundle/JobParameters/FormModelTransformerProviderInterface.php [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace Pim\Bundle\ImportExportBundle\JobParameters;
+
+use Akeneo\Component\Batch\Job\JobInterface;
+use Akeneo\Component\Batch\Model\JobInstance;
+
+/**
+ * Define form model transformer to use to edit a JobParameters depending on the Job we want configure
+ *
+ * @author    Julien <juien@akeneo.com>
```
Comment (juliensnz) [inline src/Pim/Bundle/ImportExportBundle/JobParameters/FormModelTransformerProviderInterface.php]: julien sanchez

File: src/Pim/Bundle/ImportExportBundle/Form/Type/JobParametersType.php [hunk 18]
```diff
@@ -105,6 +112,34 @@ function (FormEvent $event) use ($factory, $configProviderRegistry, $constraintP
...
+                $job                      = $this->jobRegistry->get($jobInstance->getJobName());
+                $modelTransformerProvider = $this->modelTransformerProviderRegistry->get($job);
+                $configProvider           = $this->configProviderRegistry->get($job);
+                $configs                  = $configProvider->getFormConfiguration();
+
+                $modelTransformers = (null !== $modelTransformerProvider) ?
+                    $modelTransformerProvider->getFormModelTransformers($jobInstance) :
+                    [];
+
+                $data = $event->getData();
```
Comment (juliensnz) [inline src/Pim/Bundle/ImportExportBundle/Form/Type/JobParametersType.php]: if filter 💩

File: src/Pim/Bundle/InstallerBundle/Resources/fixtures/icecat_demo_dev/jobs.yml [hunk 19]
```diff
@@ -23,11 +23,6 @@ jobs:
         label:     Demo CSV product export
         type:      export
         configuration:
```
Comment (juliensnz) [inline src/Pim/Bundle/InstallerBundle/Resources/fixtures/icecat_demo_dev/jobs.yml]: add fixtures

File: src/Pim/Component/Connector/Job/JobParameters/DefaultValuesProvider/ProductCsvExport.php [hunk 20]
```diff
@@ -39,16 +39,7 @@ public function getDefaultValues()
...
-        $parameters['locales'] = [];
-        $parameters['enabled'] = 'enabled';
-        $parameters['updated_since_strategy'] = 'all';
-        $parameters['updated_since_date'] = null;
-        $parameters['updated_since_n_days'] = null;
-        $parameters['families'] = null;
-        $parameters['categories'] = [];
-        $parameters['completeness'] = 'at_least_one_complete';
-        $parameters['product_identifier'] = null;
+        $parameters['filters'] = ['data' => [], 'structure' => (object) []];
```
Comment (juliensnz) [inline src/Pim/Component/Connector/Job/JobParameters/DefaultValuesProvider/ProductCsvExport.php]: new stdClass()

File: src/Pim/Component/Connector/Job/JobParameters/DefaultValuesProvider/ProductXlsxExport.php [hunk 21]
```diff
@@ -39,17 +39,8 @@ public function getDefaultValues()
...
-        $parameters['enabled'] = 'enabled';
-        $parameters['updated_since_strategy'] = 'all';
-        $parameters['updated_since_date'] = null;
-        $parameters['updated_since_n_days'] = null;
-        $parameters['completeness'] = 'at_least_one_complete';
-        $parameters['categories'] = [];
         $parameters['linesPerFile'] = 10000;
-        $parameters['families'] = null;
-        $parameters['product_identifier'] = null;
+        $parameters['filters'] = ['data' => [], 'structure' => []];
```
Comment (juliensnz) [inline src/Pim/Component/Connector/Job/JobParameters/DefaultValuesProvider/ProductXlsxExport.php]: new StdClass

File: src/Pim/Component/Connector/Reader/Database/ProductReader.php [hunk 22]
```diff
@@ -95,28 +79,18 @@ public function __construct(
...
-                array_filter(explode(',', $parameters->get('families')))
-            ),
-            $this->getCompletenessFilters($parameters),
-            $this->getProductIdentifiersFilter($parameters),
-            $this->getCategoryFilters($parameters)
-        );
+        $pqb = $this->pqbFactory->create(['default_scope' => $channel->getCode()]);
 
-        foreach ($filters as $filter) {
+        $filters = $parameters->get('filters');
```
Comment (nidup) [inline src/Pim/Component/Connector/Reader/Database/ProductReader.php]: Way more clean ❤

**Pull request merged**